envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt

```diff
@@ -54,6 +69,27 @@ private fun getCommunicationMode(proto: Value?): CommunicationMode {
     }
 }
 
+fun Value?.toStatusCodeFilter(properties: AccessLogFilterProperties): AccessLogFilterSettings.StatusCodeFilter? {
+    val value = if (this?.stringValue != null) this.stringValue.toLowerCase() else return null
+
+    if (!properties.statusCodeFilterPattern.matches(value)) {
+        throw NodeMetadataValidationException(
+            "Invalid access log status code filter. Expected OPERATOR:STATUS_CODE"
+        )
+    }
+    val split = value.split(properties.delimiter)
+    val op = when (split[0]) {
+        "le" -> ComparisonFilter.Op.LE
+        "eq" -> ComparisonFilter.Op.EQ
+        "ge" -> ComparisonFilter.Op.GE
+        else -> throw NodeMetadataValidationException("Invalid operator. Allowed values: le,eq,ge")
+    }
+    return AccessLogFilterSettings.StatusCodeFilter(
+        comparisonOP = op,
+        comparisonCode = split[1].toInt()
```

Comment:
This line might throw null pointer if split return a list with 1 element.

Reply:
If checked above pattern will match, split would not return 1 element, regardless of that should I catch such case?

Reply:
What if I will change pattern and it won't return 2 element list? Also pattern property IMO should be configurable to any pattern I want but know it support only `eq|le|ge:d+{3}` so maybe it shouldn't be configurable ?

Reply:
I would not make pattern configurable since it is tightly bound to the code here.

Reply:
in new commit I removed hardcoded condition. Please refer, is it better right now?